buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/classycle/Classycle.kt

```diff
@@ -72,12 +71,10 @@ open class Classycle : DefaultTask() {
     val analysisFile: File
         get() = File(reportDir, "${reportName}_analysis.xml")
 
-
-    @get:Inject
-    protected
-    open val antBuilder: IsolatedAntBuilder
-        get() = throw UnsupportedOperationException()
-
+    init {
+        this.excludePatterns = project.objects.listProperty<String>().apply { set(excludePatterns) }
+        this.reportResourcesZip = newInputFile().apply { set(reportResourcesZip) }
```

Comment:
Why do we need to create new properties which are then set to the providers? Can't we just keep a reference to them? I.e. let them be a `val` in the constructor?

Reply:
For `excludePatterns`, you're right.
For `reportResourcesZip`, I thought we needed the `RegularFileProperty` to be created via `newInputFile()` to set up the task dependencies, right?